lib/ansible/modules/system/sysctl.py

```diff
@@ -386,9 +386,21 @@ def main():
             ignoreerrors = dict(default=False, type='bool'),
             sysctl_file = dict(default='/etc/sysctl.conf', type='path')
         ),
-        supports_check_mode=True
+        supports_check_mode=True,
+        required_if=[('state', 'present', ['value'])],
     )
 
+    if module.params['name'] is None:
+        module.fail_json(msg="name can not be None")
```

Comment:
@dagwieers Thanks for your review. I added this check, because I am facing weird a behaviour, correct me if I am wrong.

Here is demo module which contains `name` and `mode` as required parameters.
When I use, ansible-playbook command with following yaml file

```yaml
- name: sample demo
   demo:
      name:
      mode:
```
I am getting `name` and `mode` as None.

Same case goes with following - 

```# ansible -m demo -a "name= mode=" localhost```

here, I am getting `name` and `mode` as "" (blank value)  
Let me know, if these behaviours are different from the normal ones. This is only reason, I have added following two checks in `sysctl` modules, because `sysctl` was writing blank values to given `sysctl.conf` file like the issue describes.